### PR TITLE
ApplicationInsights output can use custom configuration file

### PR DESCRIPTION
**Pull request by karolz-ms**

### CODE REVIEW

File: src/Microsoft.Diagnostics.EventFlow.Outputs.ApplicationInsights/ApplicationInsightsOutput.cs [hunk 1]
```diff
@@ -124,15 +126,28 @@ private void Initialize(ApplicationInsightsOutputConfiguration aiOutputConfigura
...
-            if (string.IsNullOrWhiteSpace(aiOutputConfiguration.InstrumentationKey))
+            if (string.IsNullOrWhiteSpace(aiOutputConfiguration.ConfigurationFilePath))
             {
-                string errorMessage = $"{nameof(ApplicationInsightsOutput)}: Application Insights instrumentation key is is not set)";
-                this.healthReporter.ReportProblem(errorMessage, EventFlowContextIdentifiers.Configuration);
-                throw new Exception(errorMessage);
-            }
+                if (string.IsNullOrWhiteSpace(aiOutputConfiguration.InstrumentationKey))
+                {
+                    string errorMessage = $"{nameof(ApplicationInsightsOutput)}: Application Insights instrumentation key is is not set)";
```
Comment (ammarZaidi) [inline src/Microsoft.Diagnostics.EventFlow.Outputs.ApplicationInsights/ApplicationInsightsOutput.cs]: typo: "is is"

File: src/Microsoft.Diagnostics.EventFlow.Outputs.ApplicationInsights/Configuration/ApplicationInsightsOutputConfiguration.cs [hunk 2]
```diff
@@ -10,5 +10,6 @@ namespace Microsoft.Diagnostics.EventFlow.Configuration
     public class ApplicationInsightsOutputConfiguration: ItemConfiguration
     {
         public string InstrumentationKey { get; set; }
+        public string ConfigurationFilePath { get; set; }
```
Comment (ammarZaidi) [inline src/Microsoft.Diagnostics.EventFlow.Outputs.ApplicationInsights/Configuration/ApplicationInsightsOutputConfiguration.cs]: are you going to add the description for this addition in read.md and explain the format for the file?

File: src/Microsoft.Diagnostics.EventFlow.Outputs.ApplicationInsights/ApplicationInsightsOutput.cs (hunk 1 above)
Comment (karolz-ms) [inline src/Microsoft.Diagnostics.EventFlow.Outputs.ApplicationInsights/ApplicationInsightsOutput.cs]: Thanks, will fix it before submitting

File: src/Microsoft.Diagnostics.EventFlow.Outputs.ApplicationInsights/Configuration/ApplicationInsightsOutputConfiguration.cs (hunk 2 above)
Comment (karolz-ms) [inline src/Microsoft.Diagnostics.EventFlow.Outputs.ApplicationInsights/Configuration/ApplicationInsightsOutputConfiguration.cs]: Yes, but that will be a separate change. Let me just wrap this up and publish on NuGet to get yourself unblocked.

**Pull request merged**